cocotb/handle.py

```diff
@@ -582,6 +586,10 @@ def __init__(self, handle, path):
     def __setitem__(self, index, value):
         """Provide transparent assignment to indexed array handles."""
         self[index].value = value
+        warnings.warn(
+            "Setting values on handles using the ``dut.handle[i] = value`` syntax is deprecated."
+            "Instead use the ``handle[i].value = value`` or ``handle[i] <= value`` syntax",
+            DeprecationWarning, stacklevel=2)
```

Comment:
This should happen before the write occurs, not after.

Reply:
Damn, I thought I fixed that...

Reply:
You did, but only for attr not index

Reply:
Resolved in #2509.